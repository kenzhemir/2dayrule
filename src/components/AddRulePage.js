import React, { useState, useEffect } from "react";
import { startAddRule } from "../redux/rules";
import RulesForm from "./RulesForm";
import { connect } from "react-redux";

export const AddRulePage = props => {
	const [message, setMessage] = useState("");
	useEffect(() => {
		const timer = setTimeout(() => {
			setMessage("");
		}, 3000);
		return () => {
			clearTimeout(timer);
		};
	}, [message]);
	const submitHandler = data => {
		props.addRule(data).then(() => {
			setMessage("Data is saved!");
		});
	};
	return (
		<div>
			<h3>Add new Rule</h3>
			<p>{message}</p>
			<RulesForm submitHandler={submitHandler} />
		</div>
	);
};

const mapDispatchToProps = dispatch => ({
	addRule: rule => dispatch(startAddRule(rule))
});

export default connect(undefined, mapDispatchToProps)(AddRulePage);
