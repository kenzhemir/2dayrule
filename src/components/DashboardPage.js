import React from "react";
import { connect } from "react-redux";
import { getVisibleRules } from "../redux/rules";
import { syncRules, unsyncRules, getSyncStatus } from "../redux/rulesSync";

const DashboardPage = props => (
	<>
		Your rules:
		<div>
			Listen to add is on: {props.syncStatus.toString()}
			<button
				onClick={() => {
					if (!props.syncStatus) {
						props.syncRules();
					} else {
						props.unsyncRules();
					}
				}}
			>
				turn sync {props.syncStatus ? "off" : "on"}
			</button>
		</div>
		<div>
			{props.rules.map(rule => (
				<div key={rule.id}>
					<h3>{rule.name}</h3>
					<p>{rule.description}</p>
				</div>
			))}
		</div>
	</>
);

const mapStateToProps = state => ({
	rules: getVisibleRules(state.rules),
	syncStatus: getSyncStatus(state)
});
const mapDispatchToProps = dispatch => ({
	syncRules: () => dispatch(syncRules()),
	unsyncRules: () => dispatch(unsyncRules())
});

export default connect(mapStateToProps, mapDispatchToProps)(DashboardPage);
