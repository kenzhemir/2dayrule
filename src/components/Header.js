import { connect } from "react-redux";
import { Link } from "react-router-dom";
import React from "react";

import { startLogout } from "../actions/auth";

export const Header = ({ startLogout }) => (
	<header className="header">
		<div className="content-container">
			<div className="header__content">
				<Link className="header__title" to="/dashboard">
					<h1>Boilerplate</h1>
				</Link>
				<button onClick={startLogout} class="button button--link">
					Logout
				</button>
			</div>
		</div>
	</header>
);
const mapDispatchToProps = dispatch => ({
	startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);
