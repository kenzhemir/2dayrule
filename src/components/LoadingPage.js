import React from "react";

export const LoadingPage = () => {
	return (
		<div className="loader">
			<img src="/images/loader.gif" className="loader__image" />
		</div>
	);
};

export default LoadingPage;
