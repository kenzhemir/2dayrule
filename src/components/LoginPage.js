import { connect } from "react-redux";
import React from "react";

import { startLogin } from "../redux/auth";

export const LoginPage = ({ startLogin }) => {
	return (
		<div className="box-layout">
			<div className="box-layout__box">
				<h1 className="box-layout__title">Boilerplate</h1>
				<p>Tagline for app</p>
				<button onClick={startLogin} className="button">
					Log in with Google
				</button>
			</div>
		</div>
	);
};

const mapDispatchToProps = dispatch => ({
	startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
