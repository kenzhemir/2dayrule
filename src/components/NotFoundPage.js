import { Link } from "react-router-dom";
import React from "react";

const NotFoundPage = () => (
	<div>
		404 - <Link to="/">Home</Link>
	</div>
);

export default NotFoundPage;
