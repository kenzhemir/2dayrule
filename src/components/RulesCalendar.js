import React, { cloneElement } from "react";
import moment from "moment";

moment.updateLocale("en", {
	week: {
		dow: 1
	}
});

const RulesCalendar = () => {
	let number_of_weeks = 5;
	const doneDays = [
		moment("2020-02-24"),
		moment("2020-02-28"),
		moment("2020-03-13"),
		moment("2020-03-15"),
		moment("2020-03-07")
	];
	const calendar = [];
	let date = moment()
		.startOf("month")
		.startOf("week");
	const doneDaysIndexes = doneDays
		.map(val => val.diff(date, "days"))
		.filter(val => val >= 0)
		.sort((a, b) => a - b);
	let todayIndex = moment().diff(date, "days");
	console.log(doneDaysIndexes);
	for (let i = 0, doneIndex = 0, dayIndex = 0; i < number_of_weeks; i++) {
		let week = [];
		for (let i = 0; i < 7; i++) {
			const done = doneDaysIndexes[doneIndex] == dayIndex;
			const icon = todayIndex > dayIndex ? (done ? "ðŸŸ¢" : "ðŸ”´") : undefined;
			week.push({ date, icon });
			date = date.clone().add(1, "day");
			dayIndex++;
			if (done || doneDaysIndexes[doneIndex] < dayIndex) doneIndex++;
		}
		calendar.push(week);
	}
	console.log(calendar);
	console.log(doneDaysIndexes);
	const weekdays = moment.weekdaysShort(true);
	return (
		<div className="calendar">
			<div className="calendar__title">The two day rule</div>
			<div className="calendar__weekday-row">
				{weekdays.map(weekday => (
					<div className="calendar__weekday-cell" key={weekday}>
						{weekday}
					</div>
				))}
			</div>
			<div className="calendar__month-grid">
				{calendar.map((week, weekNum) => (
					<div className="calendar__week-row" key={weekNum}>
						{week.map((day, dayNum) => (
							<div className="calendar__day-cell" key={dayNum}>
								<div className="day-cell__day">
									{day.date.format("DD")}
									<p>{day.icon}</p>
								</div>
							</div>
						))}
					</div>
				))}
			</div>
		</div>
	);
};

export default RulesCalendar;
