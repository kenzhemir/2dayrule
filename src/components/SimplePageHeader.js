import React from "react";

export const SimplePageHeader = props => {
	return (
		<div className="page-header">
			<div className="content-container">
				<h1 className="page-header__title">{props.title}</h1>
			</div>
		</div>
	);
};

export default SimplePageHeader;
