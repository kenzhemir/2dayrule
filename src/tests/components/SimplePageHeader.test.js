import { shallow } from "enzyme";
import React from "react";

import { SimplePageHeader } from "../../components/SimplePageHeader";

test("should render correctly", () => {
	const wrapper = shallow(<SimplePageHeader title="Simple title" />);
	expect(wrapper).toMatchSnapshot();
});
