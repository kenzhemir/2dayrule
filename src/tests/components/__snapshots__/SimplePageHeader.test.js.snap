// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Simple title
    </h1>
  </div>
</div>
`;
